src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -365,7 +366,9 @@ static void PromoteCommonCommitCharactersOntoList(LSP.VSCompletionList completio
                     var commitCharacters = completionItem.CommitCharacters;
                     if (commitCharacters == null)
                     {
-                        continue;
+                        // The commit characters on the item are null, this means the commit characters are actually
```

Comment:
This fixes two issues
1.  We would hit a null ref on line 384 if all of the completion items had default commit characters (aka the commit characters field on every completion item was null). 
2.  The algorithm is trying to promote the most used commit character set onto the list itself, then null out commit chars on any completion item using that set to save serialization.  However this skipped _originally_ null commit chars (indicating that the item should use commit chars from `Initialize`).  This resulted in the _second most used_ commit char set being promoted to the `CompletionList.CommitChars`, and both items with default chars and second most used commit chars having their commit chars nulled out.  This means for items that originally had null commit chars (aka use default from `Initialize`), the client would use the wrong commit chars from the completion list.

Reply:
> resulted in the second most used commit char set being promoted

lol 🤦‍♂️

Good find!